Source/WebCore/Modules/applepay/ApplePaySessionPaymentRequest.h

```diff
@@ -167,7 +167,38 @@ class ApplePaySessionPaymentRequest {
     const std::optional<ApplePayLaterAvailability>& applePayLaterAvailability() const { return m_applePayLaterAvailability; }
     void setApplePayLaterAvailability(const std::optional<ApplePayLaterAvailability>& applePayLaterAvailability) { m_applePayLaterAvailability = applePayLaterAvailability; }
 #endif
-    
+
+#if PLATFORM(COCOA)
+
+    ApplePaySessionPaymentRequest(String countryCode, String currencyCode, struct ContactFields requiredBillingContactFields, PaymentContact billingContact, ContactFields requiredShippingContactFields, PaymentContact shippingContact, Vector<String> supportedNetworks, struct MerchantCapabilities merchantCapabilities, ApplePaySessionPaymentRequestShippingType shippingType, Vector<ApplePayShippingMethod> shippingMethods, Vector<ApplePayLineItem> lineItems, ApplePayLineItem total, String applicationData, Vector<String> supportedCountries, Requester requester, PaymentInstallmentConfiguration installmentConfiguration, std::optional<ApplePayShippingContactEditingMode> shippingContactEditingMode, std::optional<bool> supportsCouponCode, String couponCode, std::optional<ApplePayRecurringPaymentRequest> recurringPaymentRequest, std::optional<ApplePayAutomaticReloadPaymentRequest> automaticReloadPaymentRequest, std::optional<Vector<ApplePayPaymentTokenContext>> multiTokenContexts, std::optional<ApplePayDeferredPaymentRequest> deferredPaymentRequest, std::optional<ApplePayLaterAvailability> applePayLaterAvailability)
```

Comment:
Most or all of these parameters should be r-value references, and they should be WTFMoved into place in the initializers.  That prevents unnecessary copies.

Reply:
Fixed